src/faivor/metrics_api.py

```diff
@@ -0,0 +1,600 @@
+            except Exception as e:
+                all_metrics[f"{category_name}.{metric.regular_name}"] = f"Error: {str(e)}"    
+    
+    def calculate_metrics(self) -> Dict[str, Any]:
+        """
+        Calculate metrics for the model.
+        
+        Returns
+        -------
+        dict
+            Dictionary containing calculated metrics.
+        """
+        y_true, y_pred, valid_indices = self.prepare_data()
+        
+        if len(y_true) == 0:
+            return {"error": "No valid numeric data pairs found for metric calculation"}
+            
+        model_type = self.metadata_json.get("model_type", "regression")
+        
+        # sensitive attribute
+        sensitive_values = self.get_sensitive_values(valid_indices)
+        feature_importance = self.metadata_json.get("feature_importance")
+        
+        all_metrics = {}
+        
+        # detect appropriate metrics module based on model type
+        metrics_module = (classification_metrics if model_type.lower() == "classification" 
+                        else regression_metrics)
+        
+        # calc metrics for each category
+        self._compute_metrics_for_category(
+            metrics_module.performance, "performance", 
+            y_true, y_pred, all_metrics
+        )
+        
+        self._compute_metrics_for_category(
+            metrics_module.fairness, "fairness", 
+            y_true, y_pred, all_metrics, 
+            sensitive_values=sensitive_values
+        )
+        
+        self._compute_metrics_for_category(
+            metrics_module.explainability, "explainability", 
+            y_true, y_pred, all_metrics, 
+            feature_importance=feature_importance
+        )
+                
+        return all_metrics    
+
+    def preprocess_probability_values(self, y_pred: np.ndarray, verbose: bool = False) -> Tuple[np.ndarray, str]:
+        """
+        Preprocess model outputs to ensure they are valid probabilities in [0,1] range.
+        
+        Parameters
+        ----------
+        y_pred : np.ndarray
+            The prediction values to process
+        verbose : bool, optional
+            Whether to print information about the transformation, by default False
+            
+        Returns
+        -------
+        Tuple[np.ndarray, str]
+            The preprocessed probabilities and a message describing the transformation
+        """
+        min_val = np.min(y_pred)
+        max_val = np.max(y_pred)
+        
+        # check if values are already in valid prob ([0,1]) range
+        if min_val >= 0 and max_val <= 1:
+            return y_pred, "No transformation needed"
+        
+        #### treatment type #1 1: If values look like logits (large range, both positive and negative)
+        if min_val < -1 or max_val > 2:
+            if verbose:
+                logger.log(f"Values range [{min_val:.4f}, {max_val:.4f}] looks like logits, applying sigmoid")
```

Comment:
```suggestion
                logger.info(f"Values range [{min_val:.4f}, {max_val:.4f}] looks like logits, applying sigmoid")
```
there was an issue detected by `pytype`, this would resolve it. Hopefully after that, the GitHub action will pass